dom-element.js

```diff
@@ -27,7 +27,7 @@ function DOMElement(tagName, owner, namespace) {
     this._attributes = {}
 
     if (this.tagName === 'INPUT') {
-      this.type = 'text'
+      this.setAttribute('type', 'text')
```

Comment:
:-1: Please do both or only properties.

Properties are source of truth, attributes are just a weird bag of stuff.